lib/gooddata/lcm/actions/provision_clients.rb

```diff
@@ -38,28 +39,37 @@ def call(params)
           domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
 
           synchronize_projects = []
-          results = params.segments.pmap do |segment|
-            tmp = domain.provision_client_projects(segment.segment_id).pmap do |m|
-              Hash[m.each_pair.to_a].merge(type: :provision_result)
-            end
 
-            unless tmp.empty?
-              synchronize_projects << {
-                segment_id: segment.segment_id,
-                from: segment.development_pid,
-                to: tmp.map do |entry|
-                  unless entry[:project_uri]
-                    raise "Provisioning project for client id #{entry[:id]} has error: #{entry[:error]}"
+          begin
+            results = params.segments.pmap do |segment|
+              tmp = domain.provision_client_projects(segment.segment_id).pmap do |m|
+                Hash[m.each_pair.to_a].merge(type: :provision_result)
+              end
+
+              unless tmp.empty?
+                synchronize_projects << {
+                  segment_id: segment.segment_id,
+                  from: segment.development_pid,
+                  to: tmp.map do |entry|
+                    unless entry[:project_uri]
+                      raise "Provisioning project for client id #{entry[:id]} has error: #{entry[:error]}"
+                    end
+                    {
+                      pid: entry[:project_uri].split('/').last,
+                      client_id: entry[:id]
+                    }
                   end
-                  {
-                    pid: entry[:project_uri].split('/').last,
-                    client_id: entry[:id]
-                  }
-                end
-              }
-            end
+                }
+              end
 
-            tmp
+              tmp
+            end
+          rescue => e
+            params.gdc_logger.error "Problem occurs when provisioning clients. Purge all invalid clients now ..."
+            res = PurgeClients.send(:call, params)
```

Comment:
Using `send` for ordinary method calls is quite non-standard. However, it's a minor issue I am willing to live with.